### DISCUSSION TITLE
Blocky mapping DNS for git commands incorrectly?

### DISCUSSION THREAD

**Discussion by adamshand** [0xERR0R/blocky#298] [Q&A]
Hello,

Just started using Blocky and it's great, thanks!  I'm running into an odd problem which I assume is a configuration error on my part, but I can't quite figure out what I'm doing wrong.

I'm doing a custom mapping for my home network.

```
customDNS:
  mapping:
    xxx.nz: 172.16.1.50
```

When I do DNS resolution on the command line everything works as expected:

```
❯ ping lakdjf.xxx.nz
PING lakdjf.xxx.nz (172.16.1.50): 56 data bytes
64 bytes from 172.16.1.50: icmp_seq=0 ttl=64 time=3.240 ms

❯ ping github.com
PING github.com (52.64.108.95): 56 data bytes
64 bytes from 52.64.108.95: icmp_seq=0 ttl=50 time=78.952 ms
```

However when I try and work with a Git repository something is remapping the DNS for `github.com` to `github.com.xxx.nz`:

```
❯ git remote -v
origin	git@github.com:adam/ansible.git (fetch)
origin	git@github.com:adam/ansible.git (push)

❯ git pull
The authenticity of host 'github.com.xxx.nz (172.16.1.50)' can't be established.
ECDSA key fingerprint is SHA256:uJJJxxxgd1sUGUP6xxx1V/gs8jhQ5+YcVPukWT5w.
No matching host key fingerprint found in DNS.
Are you sure you want to continue connecting (yes/no/[fingerprint])?
```

This is what I see in the logs:

```
blocky    | [2021-09-29 12:04:22]  INFO query_logging_resolver: query resolved answer= client_ip=172.16.1.202 client_names=172.16.1.202 duration_ms=0 question=AAAA (github.com.xxx.nz.) response_code=NXDOMAIN response_reason=CUSTOM DNS
blocky    | [2021-09-29 12:04:22]  INFO query_logging_resolver: query resolved answer= client_ip=172.16.1.202 client_names=172.16.1.202 duration_ms=0 question=SSHFP (github.com.xxx.nz.) response_code=NXDOMAIN response_reason=CUSTOM DNS
blocky    | [2021-09-29 12:04:24]  INFO query_logging_resolver: query resolved answer= client_ip=172.16.1.202 client_names=172.16.1.202 duration_ms=0 question=SSHFP (github.com.xxx.nz.) response_code=NXDOMAIN response_reason=CUSTOM DNS
```

Any suggestions would be much appreciated.
Thanks,
Adam.

**Comment by 0xERR0R** [comment]
Hi,

This looks interesting. Which operating system? I found something similar here: https://serverfault.com/questions/74067/windows-appending-domain-suffix-to-all-lookups

**Reply by adamshand** [reply]
Blocky is running inside Docker on a Debian Linux server.  I'm seeing the problem both on the Docker host and on my Mac laptop.

Interesting link, I'm pretty sure I'm not using a DNS server with a wildcard entry, but just to be sure set 1.1.1.1 as my only DNS server in Blocky and was still having the problem.

I just changed the Blocky config to add specific hosts and it seems to work as expected:

```
customDNS:
  mapping:
    books.xxx.nz: 172.16.1.50
    media.xxx.nz: 172.16.1.50
    music.xxx.nz: 172.16.1.50
```

But confused why the previous config caused problems?  Especially since I couldn't replicate it with ping/host/dig.

Thanks,
Adam.

**Comment by 0xERR0R** [comment]
The problem is that blocky receives a request "github.com.xxx.nz.". Since you configured mapping for "*.xxx.nz", blocky sends a response with the IP address as expected.

Your client shouldn't try to ask for "github.com.xxx.nz." if you want to query for "github.com".

With new configured mapping, the query "github.com.xxx.nz." will be answered with NXDOMAIN (not found) and your client uses "github.com".

Which operation system do you use with your git client?
